Pegasus/include/Math.hpp

```diff
@@ -1,26 +1,655 @@
+            {
+                m_current = m_current->next;
+                return *this;
+            }
+
+            FaceIterator operator++(int)
+            {
+                FaceIterator result(*this);
+                operator++();
+                return result;
+            }
+
+            FaceIterator& operator--()
+            {
+                m_current = m_current->prev;
+                return *this;
+            }
+
+            FaceIterator operator--(int)
+            {
+                FaceIterator result(*this);
+                operator++();
+                return result;
+            }
+
+            bool operator==(FaceIterator const& other) const
+            {
+                return m_current == other.m_current;
+            }
+
+            bool operator!=(FaceIterator const& other) const
+            {
+                return !operator==(other);
+            }
+
+            friend void swap(FaceIterator& lhs, FaceIterator& rhs) noexcept
+            {
+                std::swap(lhs.m_current, lhs.m_current);
+            }
+
+            operator FaceIterator<FaceType const>() const
+            {
+                return FaceIterator(*this);
+            }
+
+        private:
+            HalfEdge* m_current;
+        };
+
+        /**
+         * @brief Constructs a face from a given half-edge
+         * @param halfEdge face hald-edge pointer
+         */
+        explicit Face(HalfEdge* halfEdge = nullptr);
+
+        /**
+         * @brief Faces half-edge iterator getter
+         * @return half-edge iterator
+         */
+        edge_iterator GetHalfEdgeIterator();
+
+        /**
+        * @brief Faces half-edge const iterator getter
+        * @return half-edge const iterator
+        */
+        const_edge_iterator GetHalfEdgeIterator() const;
+
+        /**
+         * @brief Faces adjacent face iteratir getter
+         * @return adjacent face iterator
+         */
+        face_iterator GetAdjacentFaceIterator();
+
+        /**
+        * @brief Faces adjacent face const iteratir getter
+        * @return adjacent face const iterator
+        */
+        const_face_iterator GetAdjacentFaceIterator() const;
+
+    private:
+        HalfEdge* m_halfEdge;
+    };
+
+    /**
+     * @brief Face key for associative containers.
+     */
+    struct FaceVertices
+    {
+        uint64_t a;
+        uint64_t b;
+        uint64_t c;
+
+        bool operator==(FaceVertices const& other) const;
+    };
+
+    /**
+     * @brief FaceVertices hasher.
+     */
+    struct FaceVerticesHash
+    {
+        size_t operator()(FaceVertices const& face) const;
+    };
+
+    /**
+     * @brief Data structure that containes a half of the edge.
+     */
+    struct HalfEdge
+    {
+        HalfEdge* next;
+        HalfEdge* prev;
+        HalfEdge* twin;
+        Face* face;
+        uint64_t vertexIndex;
+    };
+
+    /**
+     * @brief Inserts face into current data structure.
+     * @param a new face vertex index
```

Comment:
Shorten to `vertex index` because it is not an index within face, it's global index that shall be used by the face.

Write regular description like `Creates new face using provided points and inserts it into current data structure`

Also check several next methods